aliases.go

```diff
@@ -44,20 +49,41 @@ func (client *Client) CreateAlias(input AliasCreateInput) ([]string, error) {
 	return output, HandleErrors(err, m.Payload.Errors)
 }
 
+func (client *Client) DeleteDomainAlias(alias string) error {
```

Comment:
This seems like a different paradigm then what was built for `TaggableResource` where you pass in the client to a function on the interface.  Why 2 paradigms?

Reply:
There is a `client.DeleteAlias()` and a `client.DeleteTag()` - but only DeleteAlias had its own wrapper functions like `DeleteServiceAlias` from 2021 and `DeleteInfraAlias` from 2023. I saw 2 paradigms to begin with and ran with the DeleteAlias funcs to match what already existed. We can walk these back and mark the wrapper Delete funcs as deprecated